Hard/NQueens_shuna.java

```diff
@@ -0,0 +1,101 @@
+import java.util.*;
+
+/**
+ * N-Queens
+ * The n-queens puzzle is the problem of placing n queens on an n×n chessboard such that no two queens attack each other.
+ * Given an integer n, return all distinct solutions to the n-queens puzzle.
+ * Each solution contains a distinct board configuration of the n-queens' placement, where 'Q' and '.' both indicate a queen and an empty space respectively.
+ * For example,
+ * There exist two distinct solutions to the 4-queens puzzle:
+ *
+ * [
+ * [".Q..",  // Solution 1
+ * "...Q",
+ * "Q...",
+ * "..Q."],
+ * ["..Q.",  // Solution 2
+ * "Q...",
+ * "...Q",
+ * ".Q.."]
+ * ]
+ * Tags:  Backtracking
+ * Similar Problems:(H) N-Queens II
+ */
+
+public class NQueens {
+    public static List<List<String>> solveNQueens(int n) {
+        List<List<String>> res = new ArrayList<List<String>>();
+        int[] board = new int[n];
+        QueenDFS(n, 0, board, res);
+        return res;
+    }
+
+    public static void QueenDFS(int nrow, int row, int[] board, List<List<String>> res){
```

Comment:
QueenDFS -> queenDfs, please follow the camel case naming convention. 
nrow is the last row or nth row, right? Might use another clearer variable name.